src/BugsnagUnity/FileManager.cs

```diff
@@ -106,30 +128,64 @@ internal static void CacheSession(string reportId)
             }
         }
 
+        internal static void CacheEvent(string reportId)
+        {
+            var eventReport = GetPendingEventReport(reportId);
+            if (eventReport == null)
+            {
+                return;
+            }
+            using (var stream = new MemoryStream())
+            using (var reader = new StreamReader(stream))
+            using (var writer = new StreamWriter(stream, new UTF8Encoding(false)) { AutoFlush = false })
+            {
+                SimpleJson.SerializeObject(eventReport.GetSerialisableEventReport(), writer);
+                writer.Flush();
+                stream.Position = 0;
+                var jsonString = reader.ReadToEnd();
+                var path = EventsDirectory + "/" + eventReport.Id + ".event";
+                WriteToDisk(jsonString, path);
```

Comment:
Wouldn't it make more sense to stream straight to disk rather than streaming into a string and then writing to disk?

Reply:
Done 👍